cli/azd/ci-build.ps1

```diff
@@ -120,37 +128,20 @@ if ($IsWindows) {
         $tagsFlag += ",oneauth"
     }
     Write-Host $msg
-    $buildFlags += @(
-        "-buildmode=exe",
-        # remove all file system paths from the resulting executable.
-        # Instead of absolute file system paths, the recorded file names
-        # will begin either a module path@version (when using modules),
-        # or a plain import path (when using the standard library, or GOPATH).
-        "-trimpath",
-        $tagsFlag,
-        # -extldflags=-Wl,--high-entropy-va: Pass the high-entropy VA flag to the linker to enable high entropy virtual addresses
-        ($ldFlag + "-linkmode=auto -extldflags=-Wl,--high-entropy-va")
-    )
 }
 elseif ($IsLinux) {
     Write-Host "Building for linux"
-    $buildFlags += @(
-        "-buildmode=pie",
-        ($tagsFlag + ",cfgo"),
-        # -extldflags=-Wl,--high-entropy-va: Pass the high-entropy VA flag to the linker to enable high entropy virtual addresses
-        ($ldFlag + "-extldflags=-Wl,--high-entropy-va")
-    )
 }
 elseif ($IsMacOS) {
     Write-Host "Building for macOS"
-    $buildFlags += @(
-        "-buildmode=pie",
-        ($tagsFlag + ",cfgo"),
-        # -linkmode=auto: Link Go object files and C object files together
-        ($ldFlag + "-linkmode=auto")
```

Comment:
We have been using cgo in the past... I don't think this is a problem because I believe (can't find much documentation evidence) that `-linkmode=auto` is default so you probably won't have problems.

Reply:
Some documentation: https://github.com/golang/go/blob/master/src/cmd/cgo/doc.go#L1085-L1089

> By default, cmd/link will decide the linking mode as follows: if the only
packages using cgo are those on a list of known standard library
packages (net, os/user, runtime/cgo), cmd/link will use internal linking
mode. Otherwise, there are non-standard cgo packages involved, and cmd/link
will use external linking mode... 

Probably a good enough heuristic.

Reply:
I spelunked into linking and it seems that the default value is [auto](https://github.com/golang/go/blob/bc5f4a555e933e6861d12edba4c2d87ef6caf8e6/src/cmd/link/internal/ld/config.go#L85)